### DISCUSSION TITLE
Hostname support required for includeIf section in gitconfig

### DISCUSSION THREAD

**Discussion by QuAzI** [git-for-windows/git#3526] [Ideas]
Something like 

```
[includeIf "hostname:github.com"]
	path = /path/to/foo.inc

[includeIf "hostname:gitlab.com"]
	path = /path/to/bar.inc

[includeIf "hostname:yourcompany.com"]
	path = /path/to/yourcompany.inc
```

It will be more useful than each repo path matching

**Comment by dscho** [comment]
First of all, it is a bit strong of a language to claim that this is "required". It is a feature request, not more, not less.

And then, the idea needs much more fleshing out: _which_ hostname are you referring to? The current machine? That won't be `github.com`, ever. The hostname of a remote? If so, which one?

It might be best if you answered those questions by implementing a proof-of-concept. It will have to touch [`include_condition_is_true()`](https://github.com/git/git/blob/v2.33.1/config.c#L294-L307), most likely.

**Reply by rimrul** [reply]
Additionally this seems to be not windows-specific and thus it should be discussed on the git mailing list.

**Comment by QuAzI** [comment]
>>  It is a feature request, not more, not less.

Sorry for that. Sometimes it is difficult to translate something properly on the fly)

>> The hostname of a remote? If so, which one?

Yeah, remote hostname. Because in general users use one account per server. As example you have
username@domain.com - in general
123456789+username@users.noreply.github.com for github. because of security settings
username@yourcompany.com - for your employer's private server
and possible different emails for gitlab, bitbucket repos.

So you have more than one git server and managing per each repo is very unusual.

@rimrul yes. You are right

**Reply by rimrul** [reply]
Jonathan Tans ["Conditional config includes based on remote URL" patch series](https://lore.kernel.org/git/cover.1634077795.git.jonathantanmy@google.com/) seems strongly related.

**Reply by dscho** [reply]
> So you have more than one git server and managing per each repo is very unusual.

Nope. Not unusual. Please do not make such claims without any evidence. Whenever you follow the "fork & open PR" flow, you have at least two remotes. That is _quite_ common.

**Reply by QuAzI** [reply]
> Nope. Not unusual. Please do not make such claims without any evidence. Whenever you follow the "fork & open PR" flow, you have at least two remotes. That is _quite_ common.

At least two remotes at the same server (in general). When you fork & open PR on github you use different email or username? In general you don't have to change something with global gitconfig per each repo.

I just want to avoid hell with `git config user.name` and `git config user.email` for half of projects

**Reply by dscho** [reply]
I was just taking exception with making unfounded claims. You have _no way_ of knowing what is very unusual for Git users or not, and therefore you cannot decide what other users need or not.
